src/main/java/iris/Parser.java

```diff
@@ -0,0 +1,134 @@
+package iris;
+
+import iris.command.AddTaskCommand;
+import iris.command.Command;
+import iris.command.DeleteTaskCommand;
+import iris.command.ExitCommand;
+import iris.command.FilterCommand;
+import iris.command.FindCommand;
+import iris.command.HelpCommand;
+import iris.command.ListCommand;
+import iris.command.MarkTaskCommand;
+import iris.command.ResetCommand;
+import iris.command.UnmarkTaskCommand;
+import iris.exception.IrisException;
+import iris.exception.MissingFieldException;
+import iris.exception.NoTaskException;
+import iris.task.Deadline;
+import iris.task.Event;
+import iris.task.Todo;
+
+/**
+ * Parses the user inputs
+ */
+public class Parser {
+    enum CommandWord {
+        help, bye, list, mark, unmark, delete, todo, deadline, event, reset, filter, find
+    }
+
+    private static void checkNotEmpty(String str, String message) throws MissingFieldException {
+        if (str.isEmpty()) {
+            throw new MissingFieldException(message);
+        }
+    }
+
+    /**
+     * parses the input of the user
+     * @param input user input
+     * @return Command corresponding to the input
+     * @throws IrisException when there is a problem with the input
+     */
+    public static Command parse(String input) throws IrisException {
+        String[] arr;
+        String name;
+        String from;
+        String to;
+        String by;
+        CommandWord command;
+        try {
+            command = CommandWord.valueOf(input.split(" ")[0]);
+        } catch (IllegalArgumentException e) {
+            throw new NoTaskException();
+        }
+        switch (command) {
+        case help:
+            return new HelpCommand();
+        case bye:
+            return new ExitCommand();
+        case list:
+            return new ListCommand();
+        case mark:
+            return new MarkTaskCommand(Integer.parseInt(input.split(" ")[1]) - 1);
+        case unmark:
+            return new UnmarkTaskCommand(Integer.parseInt(input.split(" ")[1]) - 1);
+        case delete:
+            return new DeleteTaskCommand(Integer.parseInt(input.split(" ")[1]) - 1);
+        case reset:
+            return new ResetCommand();
+        case filter:
+            try {
```

Comment:
You could make another subclass of Command to help handle filter commands like what you did above.

Reply:
Oh, I do have a subclass of Command (FilterCommand). I initiate in on line 73. Here, I'm just trying to check the input, and separate it into two cases: when we want the deadlines and events on a date and when we need all deadlines and events in a certain period (both start and end date-time is provided). 

I have different constructors for both so I need to do it here. Also, OOP-wise, I think it's the work of Parser.

Thank you for the suggestion though. Do let me know if I misunderstood your suggestion.